### ISSUE TITLE
[Feature]: reset c_agg watermark for tests

### ISSUE THREAD

**Issue by pythoneer** [timescale/timescaledb#7152]
### What problem does the new feature solve?

In our tests we don't fully recreate our database as this is to expensive and time consuming. We truncate certain tables. This conflicts with continuous aggregates as the watermark might be in the "future" for additional test runs even though the original table was truncated. There is no way to truncate or reset a continuous aggregate other than recreating it. This is not feasible as the database structure is handled by specific migrations. Having two location where the continuous aggregate is specified (migration and test fixture) would run out of sync very quickly. 

Having a function to reset the watermark `CALL reset_continuous_aggregate_watermark('c_agg_name');` would be helpful in tests to reset the internal watermark so "new" real time data is visible again after a new test run.

Our current mitigation touches internal timescale tables and in believe this is not really a long term solution as those data structure might change in the future.

```sql
        UPDATE "_timescaledb_catalog".continuous_aggs_watermark
        SET watermark = -210866803200000000
        WHERE mat_hypertable_id = (
            SELECT mat_hypertable_id FROM "_timescaledb_catalog".continuous_agg
            WHERE user_view_name = 'daily_duration')
```  

### What does the feature do?

Basically what this snipped does

```sql
        UPDATE "_timescaledb_catalog".continuous_aggs_watermark
        SET watermark = -210866803200000000
        WHERE mat_hypertable_id = (
            SELECT mat_hypertable_id FROM "_timescaledb_catalog".continuous_agg
            WHERE user_view_name = 'daily_duration')
```
as a public interface to continuous aggregates 

```sql
CALL reset_continuous_aggregate_watermark('daily_duration');
```
